ReGitLint/Cleanup.cs

```diff
@@ -111,6 +111,11 @@ public Cleanup()
             "This setting is no longer used.",
             x => DisableJbPathHack = x != null
         );
+        HasOption(
+            "long-form",
+            "Call jb command using 'dotnet tool run jb' instead of 'dotnet jb'",
+            x => LongForm = x != null
+        );
```

Comment:
Why did you do the opposite of what you proposed at https://github.com/sethreno/ReGitLint/issues/29#issuecomment-1237029356? Now you've added a breaking change for reasons I don't understand. Also, it violates semantic versioning.

Reply:
I thought it would be better for it to work "out of the box" for everything. I tested and didn't expect anything to break.
I read that short form was officially supported here https://learn.microsoft.com/en-us/dotnet/core/tools/global-tools#invoke-a-local-tool


Can you explain what broke? Also can you explain what you mean by broken semantic versioning?

Reply:
The breaking change is that ReGitLint now runs a different command, _potentially_ altering existing behavior. I haven't tried the new version yet, so I did not run into a case that actually broke. But this change means I now need to retest and verify that everything still works as it did before. Or deep-dive into the sources of the dotnet tool, to determine if it behaves exactly the same under all circumstances. So the effect of this change is I need to do extra work, without any benefit. The switch to long-form by default may introduce new issues (no software is perfect), and if we find those, will we change the default again? That's not a very sustainable approach. In my opinion, there's no reason to change the default behavior of ReGitLint (affecting all users) because of a temporary bug elsewhere that affects a small subset of users. Bugs come and go all the time, so providing a workaround to address that avoids the need to impact everyone. I therefore believe that adding `--shortForm` instead of `--longForm` is the more user-friendly, thus better solution.

Semantic versioning states that breaking changes require a major version bump. The previous behavior was in v6.2.1, while this change was introduced in v6.3.9. That's a minor version increment, therefore violating the semantic versioning requirements. Users generally expect that updating to a minor version is safe and does not require effort to adapt. While I could now choose to add `--shortForm` to my existing projects to retain the old behavior, because I know about it, most users are unaware of this change. There are no release notes and this break is not documented anywhere, as far as I'm aware.

Reply:
Sorry for any extra work this change caused. That wasn't something I anticipated. In hindsight I should have updated the issue with my thoughts before making the change so there could be more discussion.  On a related note, I'm working on adding more tests to the project. If you provide me the commands you're using in your workflow I'll add tests for them to try and help prevent regressions.

I did spend a few hours researching before I made the change and don't expect defaulting to short form to break anything. There is no downside that I know of and it has the benefit of providing a better user experience for those impacted by https://github.com/dotnet/sdk/issues/24059.

If there are actual issues defaulting to short form then I suppose I would try and determine which one would work for the majority and default to that. It doesn't appear that the underlying issue will be fixed in net6 which is still supported for 2 years.

Regarding semantic versioning I think a minor version increment was appropriate because I added a new backward compatible option. --longForm. The interface didn't change and existing commands should all still work.